packages/perseus/src/components/number-input.tsx

```diff
@@ -71,20 +70,27 @@ class NumberInput extends React.Component<any, any> {
         }
     }
 
+    _getInput: () => HTMLInputElement = () => {
+        if (!this.inputRef.current) {
+            throw new PerseusError(
+                "Input ref accessed before set",
+                Errors.Internal,
+            );
```

Comment:
Yeah, I was a little worried about doing this, but I think it's roughly the same from the learner's perspective if we throw an error intentionally or get an error from trying to access a property on an unset value.

For example, the before:
``` TS
        // @ts-expect-error - TS2531 - Object is possibly 'null'
        return ReactDOM.findDOMNode(this.refs.input).value.toString();
```

So here we'd get an error when we try to access `.value` (because I believe `findDOMNode` would return `null`). Now we'd still throw an error, but we'd be doing it intentionally and can provide an error message.

Reply:
I don't think I've seen this approach in Perseus? Can we not just do a `this.inputRef.current?.focus();` and leave it at that?

Reply:
That would require changing the logic and the return types of all the methods that weren't properly checking the ref before. I was worried that would spiral out of control if I then had to handle `undefined` where `number-input` was consumed. We do it regularly in Webapp:

``` TS
        const itemRenderer = itemRendererRef.current;
        if (itemRenderer == null) {
            // NOTE(kevinb): this should never happen since the itemRenderer
            // and the exercise chrome should be unmounted at the same time.
            throw new KAError(
                "cannot check answer on null item renderer",
                Errors.Internal,
            );
        }
        const score = itemRenderer.scoreInput();
```